### PR TITLE
Multi domain: Fix printing multi domains

### PR DESCRIPTION
**Pull request by finikorg**
Tested only inside umockdev environment.

### CODE REVIEW

**Comment by ybernat** [comment]
(Why "do not merge"? Should I close it without merging? I think you can cancel it yourself if you want.)

**Comment by finikorg** [comment]
@ybernat I marked it like this since I have tested this only inside emulated Thunderbolt environment. I want to test it also with the real "multi-domain" setup.

File: tbtadm/controller.cpp
Comment (ybernat) [inline tbtadm/controller.cpp:485]: What's the point of this change?

File: tbtadm/controller.cpp
Comment (ybernat) [inline tbtadm/controller.cpp:494]: Same here

File: tbtadm/controller.cpp
Comment (ybernat) [inline tbtadm/controller.cpp]: case 3? (and possibly more)
This need a better handling of "non SL 1" modes (see line 564)

**Review by ybernat — CHANGES_REQUESTED** [review]

File: tbtadm/controller.cpp
Comment (ybernat) [reply tbtadm/controller.cpp:485]: Oh, I see the commit description. OK, if this is what you prefer.

File: tbtadm/controller.cpp
Comment (finikorg) [reply tbtadm/controller.cpp]: I think this would be handled in the other patch series in PR #45 , I include this change because my patches based on it.

File: tbtadm/controller.cpp
Comment (finikorg) [reply tbtadm/controller.cpp:485]: Yes, otherwise we may have different output if there is no newline in the sysfs entry

Comment (ybernat) [reply tbtadm/controller.cpp:485]: There has a new line if you eep it using `read()` instead `readAndTrim()`, yes?

File: tbtadm/controller.cpp
Comment (ybernat) [reply tbtadm/controller.cpp]: Oh, so this depends on the other one. OK, let's mention it next time so it'll be clear in first place.

File: tbtadm/controller.cpp
Comment (finikorg) [reply tbtadm/controller.cpp:485]: Yes, but for simulated environment there is no newline, so this really depends on sysfs entry. We should also verify that entry is more-less correct checking size, etc.

Comment (ybernat) [reply tbtadm/controller.cpp:485]: So the simulation is bad simulation... sysfs must include a newline, as much as I know.

File: tbtadm/controller.cpp
Comment (ybernat) [inline tbtadm/controller.cpp:437]: = instead of += breaks the printing of a chain of devices

**Review by ybernat — CHANGES_REQUESTED** [review]

File: tbtadm/controller.cpp
Comment (finikorg) [reply tbtadm/controller.cpp:437]: This is AFAIK first level of indentation, so after you have added "|    " it would be hanging there forever. For the last controller we do not need it. I will add multi device controller test.

Comment (ybernat) [reply tbtadm/controller.cpp:437]: Oh, sorry, you are correct, of course. This is about the controller itself.
Adding such a test will be nice anyway :)
And this needs rebasing before merging it (and probably squashing the last commit into the one that changed `read` to `readAndTrim`).

**Comment by finikorg** [comment]
@ybernat what do you think about the last patch set?

**Comment by ybernat** [comment]
Any idea what are the question marks in https://travis-ci.org/intel/thunderbolt-software-user-space/builds/329022074#L2521 ?

**Comment by finikorg** [comment]
@ybernat No idea, I do not see those when running locally with "make check" and "make docker-run"

**Comment by finikorg** [comment]
@ybernat after rebasing with upstream I do not see that issue any more.

**Comment by ybernat** [comment]
So there is no issue anymore and the PR can be cancelled or you meant the question marks disappear?
If the latter, could you please push here the rebased commits so we can see it works?

**Comment by finikorg** [comment]
@ybernat question mark disappeared, the commits already pushed, see above.

**Pull request merged**
